src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2893,7 +2891,15 @@ private void EnsureDescriptorsInitialized()
                 EventDispatcher? dispatcher = m_Dispatchers;
                 while (dispatcher != null)
                 {
-                    dispatcher.m_EventEnabled ??= new bool[m_eventData.Length];
+                    Dictionary<int, bool> eventEnabled = new Dictionary<int, bool>();
```

Comment:
```suggestion
                    Dictionary<int, bool> eventEnabled = new Dictionary<int, bool>(m_eventData?.Count ?? 0);
```
We can pre-allocate the dictionary of the right size